File: sdks/go/tests/conformance/sink_queue_flush_shutdown_test.go
package conformance

import (
	"context"
	"testing"

	"github.com/astraive/loza/sdks/go"
)

func TestSinkQueueFlushAndShutdownHelpers(t *testing.T) {
	sink, store := loza.MemorySink()
	if err := loza.Configure(loza.Test().WithSink(sink).WithAsync(false)); err != nil {
		t.Fatalf("configure: %v", err)
	}
	if err := loza.Default().Event(context.Background(), "sink.event", loza.String("family", "sink")); err != nil {
		t.Fatalf("event: %v", err)
	}
	if err := loza.Flush(context.Background()); err != nil {
		t.Fatalf("flush: %v", err)
	}
	if store.Len() == 0 {
		t.Fatalf("expected sink event to be flushed")
	}
	if err := loza.Drain(context.Background(), sink); err != nil {
		t.Fatalf("drain: %v", err)
	}
	loza.Pause(sink)
	loza.Resume(sink)
	if got := loza.QueueSize(sink); got != 0 {
		t.Fatalf("expected empty queue, got %d", got)
	}
	if err := loza.Health(context.Background(), sink); err != nil {
		t.Fatalf("health: %v", err)
	}
	if loza.StdoutSink() == nil {
		t.Fatalf("expected stdout sink")
	}
	if loza.NoopSink() == nil {
		t.Fatalf("expected noop sink")
	}
	if loza.MultiSink(sink, loza.NoopSink()) == nil {
		t.Fatalf("expected multi sink")
	}
	if err := loza.Shutdown(context.Background()); err != nil {
		t.Fatalf("shutdown: %v", err)
	}
}

File: sdks/js/tests/sink_queue_flush_shutdown.test.ts
import { afterEach, it } from 'node:test';
import assert from 'node:assert/strict';
import * as loza from '../src/index.ts';

afterEach(() => {
  loza.reset();
});

it('covers sink, queue, flush, and shutdown helpers', async () => {
  const sink = new loza.MemorySink();
  loza.configure(loza.test('catalog').withSink(sink));
  const logger = loza.createLoza({ service: 'catalog', sink });

  logger.info('sink.event', { family: 'sink' });
  await logger.flush();
  assert.ok(sink.getLength() >= 1);

  const batchSink = loza.httpBatchSink({ endpoint: 'http://127.0.0.1:65535/events', batchSize: 10, flushIntervalMs: 1000 });
  batchSink.pause();
  batchSink.resume();
  assert.equal(batchSink.queueSize(), 0);

  assert.ok(loza.stdoutSink());
  assert.ok(loza.noopSink());
  assert.ok(loza.multiSink(sink, loza.noopSink()));
  assert.ok(loza.otlpSink('http://127.0.0.1:4318'));

  await loza.flush();
  await logger.shutdown();
  await loza.shutdown();
});
